

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Bacterial Comparative Genomics Workshop &mdash; Micro 612 genomics workshop 3.0 documentation</title>
  

  
  

  
  <link href='https://fonts.googleapis.com/css?family=Lato:400,700,400italic,700italic|Roboto+Slab:400,700|Inconsolata:400,700&subset=latin,cyrillic' rel='stylesheet' type='text/css'>

  
  
    

  

  
  
    <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  

  

  
    <link rel="top" title="Micro 612 genomics workshop 3.0 documentation" href="#"/> 

  
  <script src="https://cdnjs.cloudflare.com/ajax/libs/modernizr/2.6.2/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-nav-search">
        
          <a href="#" class="fa fa-home"> Micro 612 genomics workshop</a>
        
        
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
        
          
          
              <!-- Local TOC -->
              <div class="local-toc"><ul>
<li><a class="reference internal" href="#">Bacterial Comparative Genomics Workshop</a><ul>
<li><a class="reference internal" href="#prerequisites">Prerequisites:</a></li>
<li><a class="reference internal" href="#workshop">Workshop:</a></li>
</ul>
</li>
</ul>
</div>
          
        
      </div>
      &nbsp;
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" role="navigation" aria-label="top navigation">
        <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
        <a href="#">Micro 612 genomics workshop</a>
      </nav>


      
      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href="#">Docs</a> &raquo;</li>
      
    <li>Bacterial Comparative Genomics Workshop</li>
      <li class="wy-breadcrumbs-aside">
        
          <a href="_sources/index.txt" rel="nofollow"> View page source</a>
        
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document">
            
  <div class="section" id="bacterial-comparative-genomics-workshop">
<span id="bacterial-comparative-genomics-workshop"></span><h1>Bacterial Comparative Genomics Workshop<a class="headerlink" href="#bacterial-comparative-genomics-workshop" title="Permalink to this headline">¶</a></h1>
<p><strong><em>A 3 day microbial bioinformatics workshop conducted by <a class="reference external" href="http://thesnitkinlab.com/index.php">Dr. Evan Snitkin</a> at <a class="reference external" href="https://www.umich.edu/">University of Michigan</a>. This module covers the basics of microbial genomic analysis using publicly available tools that are commonly referenced in genomics literature. Students will learn the steps and associated tools that are required to process, annotate and compare microbial genomes.</em></strong></p>
<p><strong><em>Date: Feb 28 - 2 March 2018</em></strong></p>
<!---
Link to Software Carpentry Etherpad:
http://pad.software-carpentry.org/micro612_bacterial_genomics_workshop
--><hr class="docutils" />
<div class="section" id="prerequisites">
<span id="prerequisites"></span><h2>Prerequisites:<a class="headerlink" href="#prerequisites" title="Permalink to this headline">¶</a></h2>
<ul class="simple">
<li>Prior participation in a <a class="reference external" href="https://umswc.github.io/2018-02-26-UMich/">Software Carpentry Workshop</a></li>
</ul>
<!---
- [Micro612 pre-course hw](https://github.com/alipirani88/Comparative_Genomics/blob/master/Micro612_pre-course_hw/Micro612_w18_pre-course_hw.pdf): A pre-course homework will help setting up Micro612 flux directories and bash profile.
-->
***</div>
<div class="section" id="workshop">
<span id="workshop"></span><h2>Workshop:<a class="headerlink" href="#workshop" title="Permalink to this headline">¶</a></h2>
<p><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day1_morning/README.md">Day 1 Morning</a></p>
<hr class="docutils" />
<ul class="simple">
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day1_morning/README.md#installing-and-setting-up-cyberduck-for-file-transfer">Installing and setting up Cyberduck for file transfer</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day1_morning/README.md#getting-your-data-onto-glux-and-setting-up-environment-variable">Getting your data onto Flux and setting up Environment variable</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day1_morning/README.md#unix-is-your-friend">Unix is your friend</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day1_morning/README.md#quality-control-using-fastqc">Quality Control using FastQC</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day1_morning/README.md#quality-trimming-using-trimmomatic">Quality Trimming using Trimmomatic</a></li>
</ul>
<p><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day1_afternoon/README.md#day-1-afternoon">Day 1 Afternoon</a></p>
<hr class="docutils" />
<ul class="simple">
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day1_afternoon/README.md#read-mapping">Read Mapping</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day1_afternoon/README.md#variant-calling-and-filteration">Variant Calling</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day1_afternoon/README.md#visualize-bam-and-vcf-files-in-artemis">Visualize BAM/VCF files in Artemis</a></li>
</ul>
<p><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day2_morning/README.md#day-2-morning">Day 2 Morning</a></p>
<hr class="docutils" />
<ul class="simple">
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day2_morning/README.md#genome-assembly">Genome Assembly</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day2_morning/README.md#assembly-evaluation-using-quast">Assembly evaluation</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day2_morning/README.md#compare-assembly-to-reference-genome-and-post-assembly-genome-improvement">Compare assembly to reference genome and Post-assembly genome improvement</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day2_morning/README.md#map-reads-to-the-final-ordered-assembly">Map reads to the final ordered assembly</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day2_morning/README.md#genome-annotation">Genome Annotation</a></li>
</ul>
<p><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day2_afternoon/README.md#day-2-afternoon">Day 2 Afternoon</a></p>
<hr class="docutils" />
<ul class="simple">
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day2_afternoon/README.md#determine-which-genomes-contain-beta-lactamase-genes">Determine which genomes contain beta-lactamase genes</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day2_afternoon/README.md#identification-of-antibiotic-resistance-genes-with-ariba-directly-from-paired-end-reads">Identification of antibiotic resistance genes with ARIBA directly from paired-end reads</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day2_afternoon/README.md#perform-pan-genome-analysis-with-roary">Perform pan-genome analysis with Roary</a></li>
</ul>
<p><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day3_morning/README.md#day-3-morning">Day 3 Morning</a></p>
<hr class="docutils" />
<ul class="simple">
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day3_morning/README.md#perform-whole-genome-alignment-with-Mauve">Perform whole genome alignment with Mauve</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day3_morning/README.md#perform-some-dna-sequence-comparisons-and-phylogenetic-analysis-in-ape">Perform DNA sequence comparisons and phylogenetic analysis in ape</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day3_morning/README.md#perform-snp-density-analysis-to-discern-evidence-of-recombination">Perform SNP density analysis to discern evidence of recombination</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day3_morning/README.md#perform-recombination-filtering-with-gubbins">Perform recombination filtering with gubbins</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day3_morning/README.md#create-annotated-publication-quality-trees-with-itol">Create annotated publication quality trees with iTOL</a></li>
</ul>
<p><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day3_afternoon/README.md#day-3-afternoon">Day 3 Afternoon</a></p>
<hr class="docutils" />
<ul class="simple">
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day3_afternoon/README.md#perform-qc-on-fastq-files">Perform QC on fastq files</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day3_afternoon/README.md#examine-results-of-spandx-pipeline">Examine results of SPANDx pipeline</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day3_afternoon/README.md#recombination-detection-and-tree-generation">Recombination detection and tree generation</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day3_afternoon/README.md#phylogenetic-tree-annotation-and-visualization">Phylogenetic tree annotation and visualization</a></li>
<li><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/day3_afternoon/README.md#assessment-of-genomic-deletions">Assessment of genomic deletions</a></li>
</ul>
<p><a class="reference external" href="https://github.com/alipirani88/Comparative_Genomics/blob/master/online_resources/README.md#helpful-resources-for-microbial-genomics">Helpful resources for microbial genomics</a></p>
<hr class="docutils" />
</div>
</div>


          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018, Evan Snitkin, Ali Pirani.
    </p>
  </div>

  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/snide/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
  
</footer>
        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'./',
            VERSION:'3.0',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true
        };
    </script>
      <script type="text/javascript" src="_static/jquery.js"></script>
      <script type="text/javascript" src="_static/underscore.js"></script>
      <script type="text/javascript" src="_static/doctools.js"></script>

  

  
  
    <script type="text/javascript" src="_static/js/theme.js"></script>
  

  
  
  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.StickyNav.enable();
      });
  </script>
   

</body>
</html>